platforms/software/reporting/src/test/groovy/org/gradle/api/plugins/ReportingBasePluginTest.groovy

```diff
@@ -42,4 +42,31 @@ class ReportingBasePluginTest extends AbstractProjectBuilderSpec {
         }
     }
 
+    def "defaults to reports dir in build dir"() {
+        project.pluginManager.apply(ReportingBasePlugin)
+        def extension = project.reporting
+
+        expect:
+        extension.baseDirectory.get().asFile == new File(project.layout.buildDirectory.get().asFile, ReportingExtension.DEFAULT_REPORTS_DIR_NAME)
+
+        when:
+        project.layout.buildDirectory.set(project.file("newBuildDir"))
+
+        then:
+        extension.baseDirectory.get().asFile == new File(project.file("newBuildDir"), ReportingExtension.DEFAULT_REPORTS_DIR_NAME)
+    }
+
+    def "defaults to reports directory in build dir"() {
```

Comment:
💅 The differences between these 2 tests are somewhat confusing.  Can they be combined and parameterized?  Is the `get().asFile` vs. `.asFile.get()` change meaningful?